### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by faheel**
* Improve Markdown formatting
  * Use lists and headings at appropriate places (eg. to list configuration options/properties)
  * Increase heading levels that were too small
  * Remove extra indentation at a few places
* Use consistent formatting rules for all Markdown files
  * Files and folders are in bold
  * Code and commands are in code spans/blocks and syntax highlighted
* Fix typos
* Fix broken links

### CODE REVIEW

File: Documentation/LanguageServer/IntelliSense engine.md [hunk 1]
```diff
@@ -15,26 +15,30 @@ Ultimately all the IntelliSense and code browsing features will be powered by th
...
 
-* `"browse.path"`: This array of path strings is used by the "Tag Parser" (a.k.a. "browse engine"). This engine will **recursively** enumerate all files under the paths specified and track them as potential includes while tag parsing your project folder. To disable recursive enumeration of a path, you can append a /* to the path string.
+* #### `"browse.path"`
+  This array of path strings is used by the "Tag Parser" (a.k.a. "browse engine"). This engine will **recursively** enumerate all files under the paths specified and track them as potential includes while tag parsing your project folder. To disable recursive enumeration of a path, you can append a `/*` to the path string.
 
-#### How fallback works and how to control the behavior
+### How fallback works and how to control the behavior?
```
Comment (bobbrow) [inline Documentation/LanguageServer/IntelliSense engine.md]: I don't think that a question mark belongs here. No question is being asked.

File: Documentation/LanguageServer/FAQ.md [hunk 2]
```diff
@@ -1,27 +1,23 @@
...
-
-[How do I get the new IntelliSense to work with MinGW on Windows](#how-do-i-get-the-new-intellisense-to-work-with-mingw-on-windows)?
-
-[What is the difference between "includePath" and "browse.path" in c\_cpp\_properties.json](#what-is-the-difference-between-includepath-and-browsepath-in-c_cpp_propertiesjson)?
-
-[How do I re-create the IntelliSense database](#how-do-i-re-create-the-intellisense-database)?
+* [How do I get IntelliSense to work correctly?](#how-do-i-get-intellisense-to-work-correctly)
+* [Why do I see red squiggles under Standard Library types?](#why-do-i-see-red-squiggles-under-standard-library-types)
+* [How do I get the new IntelliSense to work with MinGW on Windows?](#how-do-i-get-the-new-intellisense-to-work-with-mingw-on-windows)
+* [What is the difference between `"includePath"` and `"browse.path"` in **c_cpp_properties.json**?](#what-is-the-difference-between-includepath-and-browsepath-in-c-cpp-propertiesjson)
```
Comment (bobbrow) [inline Documentation/LanguageServer/FAQ.md]: The link appears to be removed (link #4)

File: README.md [hunk 3]
```diff
@@ -1,13 +1,15 @@
 # vscode-cpptools
 
-This repo is the official repository for filing issues against and getting support for the Microsoft C/C++ extension for VsCode (https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools).
+This is the official repository for filing issues against and getting support for the [Microsoft C/C++ extension for VS Code](https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools).
 
 ### Microsoft Open Source Code of Conduct
-This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/). For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments. 
+
+This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/). For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact opencode@microsoft.com with any additional questions or comments.
```
Comment (bobbrow) [inline README.md:7]: any particular reason why the mailto: link is removed?

**Review by bobbrow — CHANGES_REQUESTED** [review]
Thanks for the improvements to our documentation!

File: README.md (hunk 3 above)
Comment (sean-mcmanus) [reply README.md:7]: The link seems to work without it? Maybe GitHub is smart about it and adds stuff.

Comment (faheel) [reply README.md:7]: It was redundant. A mailto link is automatically generated for an email address.

File: Documentation/LanguageServer/IntelliSense engine.md (hunk 1 above)
Comment (faheel) [reply Documentation/LanguageServer/IntelliSense engine.md]: You're right. I'll fix it.

File: Documentation/LanguageServer/FAQ.md (hunk 2 above)
Comment (faheel) [reply Documentation/LanguageServer/FAQ.md]: Fixed.

**Review by sean-mcmanus — APPROVED** [review]
Looks like a bunch of errors were fixed. And different tagging used. Seems good to me.

**Comment by sean-mcmanus** [comment]
@bobbrow Were you okay with accepting this pull request?

**Comment by bobbrow** [comment]
I'm confused.  I didn't see any updates pushed to this PR and there is only one commit listed.  When did it get updated?

**Review by bobbrow — APPROVED** [review]
:shipit:

**Comment by faheel** [comment]
I squashed the changes suggested in the review into the original PR commit.

**Pull request merged**